java/test/jmri/jmrit/display/layoutEditor/LayoutEditorChecksTest.java

```diff
@@ -1,40 +1,316 @@
+        Assert.assertEquals("resultsMenuItem0.getText(): ", myBlockName, resultsMenuItem.getText());
+
+        JMenuOperator resultsJMO = new JMenuOperator((JMenu) resultsMenuItem);
+        JPopupMenu resultsPopupMenu = resultsJMO.getPopupMenu();
+        Assert.assertEquals("Correct number of check noncontiguous blocks results submenu items",
+                2, resultsPopupMenu.getSubElements().length);
+        JMenuItem resultsSubMenuItem = (JMenuItem) resultsPopupMenu.getComponent(0);
+        Assert.assertEquals("resultsSubMenuItem0.getText(): ", myBlockName + ":#1", resultsSubMenuItem.getText());
+        resultsSubMenuItem = (JMenuItem) resultsPopupMenu.getComponent(1);
+        Assert.assertEquals("resultsSubMenuItem1.getText(): ", myBlockName + ":#2", resultsSubMenuItem.getText());
+    }   //testCheckNonContiguousBlocks
+
+    @Test
+    public void testCheckUnnecessaryAnchors() {
+        Assume.assumeFalse(GraphicsEnvironment.isHeadless());
+        Assert.assertNotNull("toolsMenuTitle null", toolsMenuTitle);
+        JMenuOperator toolsJMO = new JMenuOperator(layoutEditorEFO, toolsMenuTitle);
+        Assert.assertNotNull("CheckMenuTitle null", checkMenuTitle);
+        Assert.assertNotNull("checkUnnecessaryAnchorsMenuTitle null", checkUnnecessaryAnchorsMenuTitle);
+
+        JPopupMenu toolsPopupMenu = toolsJMO.getPopupMenu();
+        JMenuItem checkMenuItem = (JMenuItem) toolsPopupMenu.getComponent(0);
+        Assert.assertEquals("checkMenuItem.getText(): ",
+                checkMenuTitle, checkMenuItem.getText());
+
+        JMenuOperator checkJMO = new JMenuOperator((JMenu) checkMenuItem);
+        JPopupMenu checksPopupMenu = checkJMO.getPopupMenu();
+        JMenuItem checkUnnecessaryAnchorsMenuItem = (JMenuItem) checksPopupMenu.getComponent(3);
+        Assert.assertEquals("subMenuItem.getText(): ",
+                checkUnnecessaryAnchorsMenuTitle, checkUnnecessaryAnchorsMenuItem.getText());
+
+        JMenuOperator checkUnnecessaryAnchorsJMO = new JMenuOperator((JMenu) checkUnnecessaryAnchorsMenuItem);
+        checkUnnecessaryAnchorsJMO.doClick();
+
+        JPopupMenu checkUnnecessaryAnchorsPopupMenu = checkUnnecessaryAnchorsJMO.getPopupMenu();
+
+        //verify results
+        Assert.assertEquals("Correct number of check unnecessary anchors results menu items",
+                2, checkUnnecessaryAnchorsPopupMenu.getSubElements().length);
+        JMenuItem resultsMenuItem = (JMenuItem) checkUnnecessaryAnchorsPopupMenu.getComponent(0);
+        Assert.assertEquals("resultsMenuItem0.getText(): ", a1.getName(), resultsMenuItem.getText());
+        resultsMenuItem = (JMenuItem) checkUnnecessaryAnchorsPopupMenu.getComponent(1);
+        Assert.assertEquals("resultsMenuItem1.getText(): ", a2.getName(), resultsMenuItem.getText());
+    }   //testCheckUnnecessaryAnchors
+
+    @BeforeClass
+    public static void setUpClass() {
```

Comment:
@BeforeClass and @AfterClass are supposed to be used for tasks that take a significant amount of time, such as setting up a database.

Starting layout editor does NOT take a significant amount of time, even if you configure items in the layout editor.

When you only configure the layout editor once for all tests in the file, then any operation you perform in one test changes the initial state for the next test, which means the tests Katy be order dependent.

This could very well impact why some tests fail in some environment and not others.